apps/emqx/src/emqx_frame.erl

```diff
@@ -685,12 +685,24 @@ parse_utf8_pair(<<LenK:16/big, _Key:LenK/binary, LenV:16/big, Rest/binary>>, _St
         parsed_value_length => LenV,
         remaining_bytes_length => byte_size(Rest)
     });
+parse_utf8_pair(<<LenK:16/big, Key:LenK/binary, Rest/binary>>, _StrictMode) ->
+    ?PARSE_ERR(#{
+        cause => malformed_user_property_missing_value,
+        parsed_key_length => LenK,
+        parsed_key => Key,
+        remaining_bytes_length => byte_size(Rest)
+    });
 parse_utf8_pair(Bin, _StrictMode) when
     4 > byte_size(Bin)
 ->
     ?PARSE_ERR(#{
         cause => user_property_not_enough_bytes,
```

Comment:
Q: What do you think about just dropping this weird guard, replacing the cause with `malformed_user_property` and calling it a day? Introducing two extra clauses feels a little redundant to me here.